python/ray/dashboard/dashboard.py

```diff
@@ -395,7 +403,7 @@ def run(self):
         while True:
             time.sleep(1.0)
             with self._raylet_stats_lock:
-                for node, stub in zip(self.nodes, self.stubs):
+                for node, stub in zip(self.nodes, self.stubs.values()):
```

Comment:
`self.stubs.values()` may not return the stubs in the right order. We should instead iterate only over `self.nodes`, then use the node IDs to index into `self.stubs` to ensure we get the right stub.